File: Python/Problems/Problems/problem1288.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem

class Problem1288(Problem):
    def test(self) -> bool:
        intervals = [[1, 4], [3, 6], [2, 8]]

        expected = 2

        result = self.removeCoveredIntervals(intervals)

        print(result)

        return result == expected

    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0],-x[1]))
        end = -1
        result = 0
        for interval in intervals:
            if interval[1] > end:
                end = interval[1]
                result += 1

        return result

File: C++/Problems/Problems/problem_1288.cpp
#include "problem_1288.h"

bool problem_1288::test() {
    std::vector<std::vector<int>> intervals  { { 1, 4 }, { 3, 6 }, { 2, 8 } };

    int expected = 2;

    auto result = removeCoveredIntervals(intervals);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1288::removeCoveredIntervals(std::vector<std::vector<int>> &intervals) {
    std::sort(intervals.begin(), intervals.end(), [](std::vector<int> &a, std::vector<int> &b) { return a[0] == b[0] ? b[1] < a[1] : a[0] < b[0]; });
    int end = -1;
    int result = 0;
    for (auto interval : intervals)
    {
        if (interval[1] > end)
        {
            end = interval[1];
            result++;
        }
    }

    return result;
}
